pkg/buses/message_dispatcher.go

```diff
@@ -29,9 +29,14 @@ import (
 
 const correlationIDHeaderName = "Knative-Correlation-Id"
 
+// httpDoer is an interface for making HTTP requests.
+type httpDoer interface {
+	Do(*http.Request) (*http.Response, error)
```

Comment:
I think two things: 

1- if you want to put an interface infront of the http client, you need to implement all the methods:
```
type Client
    func (c *Client) Do(req *Request) (*Response, error)
    func (c *Client) Get(url string) (resp *Response, err error)
    func (c *Client) Head(url string) (resp *Response, err error)
    func (c *Client) Post(url, contentType string, body io.Reader) (resp *Response, err error)
    func (c *Client) PostForm(url string, data url.Values) (resp *Response, err error)
```

2 - the documentation says this: `Generally Get, Post, or PostForm will be used instead of Do.`

Which implies we are using it wrong and we should take a look at that.